src/backend/distributed/operations/shard_cleaner.c

```diff
@@ -680,7 +679,7 @@ InsertCleanupRecordInCurrentTransaction(CleanupObject objectType,
 
 
 /*
- * InsertCleanupRecordInSeparateTransaction inserts a new pg_dist_cleanup_record entry
+ * InsertCleanupRecordInSubtransaction inserts a new pg_dist_cleanup_record entry
```

Comment:
I see we tried to be aligned with other `InsertCleanup...` functions here. Why do we change `separate` with `subtransaction` here? When I go back to PR #6258, we have many `InsertCleanupInSubtransaction`. Isn't `separate` a proper name here? Actually, when I see our codebase, we had many similar cases in which we use 'subtransaction'. I am little bit confused with the keyword. When I see subtransaction, I thought it is something related to `savepoint` inside the same transaction block, but afaics it is actually using new connection with separate transaction. (cc @onderkalaci)

Reply:
`Subtransaction` functions use existing connections (if any) with `OUTSIDE_TRANSACTION` flag.
`SeparateTransaction` functions force new connections, which is commited right away. Doesn't rollback if the current transaction fails or rolls back.